discovery-infra/log_scrap.py

```diff
@@ -0,0 +1,216 @@
+            }
+        })
+        return res
+
+    def is_update_needed(self, cluster_id, event_list):
+        lest_logged_event_time = self.get_last_cluster_event_time(cluster_id)
+        if not lest_logged_event_time:
+            log.info(f"Cluster {cluster_id} has no record, starting new cluster update")
+            return True
+
+        last_event_time = event_list[-1]["event_time"]
+        tdelta = datetime.strptime(last_event_time, FMT) - datetime.strptime(lest_logged_event_time, FMT)
+        if tdelta == 0:
+            log.info(f"No new events for {cluster_id}, continuing")
+            return False
+        log.info(f"New events for cluster {cluster_id}")
+        return True
+
+    def get_metadata_json(self, cluster: dict):
+        d = {'cluster': cluster}
+        d.update(self.client.get_versions())
+        return d
+
+    def process_cluster(self, cluster):
+        with tempfile.NamedTemporaryFile() as temp_event_file:
+            self.write_events_file(cluster, temp_event_file.name)
+            with open(temp_event_file.name) as f:
+                event_list = json.load(f)
+
+        self.elastefy_events(cluster, event_list)
+
+    def elastefy_events(self, cluster, event_list):
+
+        cluster_id = cluster["id"]
+
+        event_count = len(event_list)
+        if event_count > MAX_EVENTS:
+            log.info(f"Cluster {cluster_id} has {event_count} event records, logging only {MAX_EVENTS}")
+            event_list = event_list[:MAX_EVENTS]
+
+        metadata_json = self.get_metadata_json(cluster)
+
+        if self.backup_destination:
+            self.save_new_backup(cluster_id, event_list, metadata_json)
+
+        cluster_bash_data = process_metadata(metadata_json)
+        event_names = get_cluster_object_names(cluster_bash_data)
+
+        for event in event_list[::-1]:
+            cluster_bash_data["no_name_message"] = get_no_name_message(event["message"], event_names)
+            process_event_doc(event, cluster_bash_data)
+            doc_id = get_doc_id(cluster_bash_data)
+            ret = self.log_doc(cluster_bash_data, doc_id)
+            if not ret:
+                break
+
+    def save_new_backup(self,cluster_id, event_list, metadata_json):
+        cluster_backup_directory_path = os.path.join(self.backup_destination, f"cluster_{cluster_id}")
+        if not os.path.exists(cluster_backup_directory_path):
+            os.makedirs(cluster_backup_directory_path)
+
+        event_dest = os.path.join(cluster_backup_directory_path, "events.json")
+        with open(event_dest, "w") as f:
+            json.dump(event_list, f, indent=4)
+
+        metadata_dest = os.path.join(cluster_backup_directory_path, "metadata.json")
+        with open(metadata_dest, "w") as f:
+            json.dump(metadata_json, f, indent=4)
+
+    def log_doc(self, doc, id_):
+        try:
+            res = self.es.create(index=self.index, body=doc, id=id_)
+        except elasticsearch.exceptions.ConflictError:
```

Comment:
What are the chances of conflicts? Note that if you use `index` instead of `create` Elasticsearch will index a "new" version of the document, instead of raising an error. This may be something to consider.